Assets/AnkrSDK/Examples/UseCases/Ads/AdsUseCaseController.cs

```diff
@@ -1,73 +1,223 @@
+		private async void CallbackListenerOnAdFinished()
+		{
+			await UniTask.SwitchToMainThread();
+			UpdateUILogs("CallbackListenerOnAdFinished");
+			ActivateNextButton(0);
+		}
 
-			if (requestResult != null)
-			{
-				await UniTask.WhenAll(
-					_ankrBannerAdImage.SetupAd(requestResult),
-					_ankrBannerAdSprite.SetupAd(requestResult));
-			}
+		private async void CallbackListenerOnAdLoaded()
+		{
+			await UniTask.SwitchToMainThread();
+			UpdateUILogs("CallbackListenerOnAdLoaded");
+			ActivateNextButton(2);
+		}
 
-			_ankrBannerAdImage.TryShow();
-			_ankrBannerAdSprite.TryShow();
+		private async void CallbackListenerOnAdOpened()
+		{
+			await UniTask.SwitchToMainThread();
+			UpdateUILogs("CallbackListenerOnAdOpened");
 		}
 
-		public override void DeActivateUseCase()
+		private async void CallbackListenerOnAdFailedToLoad()
+		{
+			await UniTask.SwitchToMainThread();
+			UpdateUILogs("CallbackListenerOnAdFailedToLoad");
+		}
+		#endregion
+
+		#region On Button Clicks
+		private void OnInitializeButtonClick()
+		{
+			const string walletAddress = "This is ankr mobile address";
+			const string appId = "1c562170-9ee5-4157-a5f8-e99c32e73cb0";
+			_ankrAdsAndroidCallbackListener = new AnkrAdsAndroidCallbackListener();
+			UnsubscribeToCallbackListenerEvents();
+			SubscribeToCallbackListenerEvents();
+			AnkrAdsNativeAndroid.Initialize(appId, walletAddress,_ankrAdsAndroidCallbackListener);
+		}
+
+		private void OnLoadButtonClick()
+		{
+			const string testUnitId = "d396af2c-aa3a-44da-ba17-68dbb7a8daa1";
+			AnkrAdsNativeAndroid.LoadAd(testUnitId);
+		}
+		
+		private void OnGetTextureClick()
+		{
+			const string testUnitId = "d396af2c-aa3a-44da-ba17-68dbb7a8daa1";
+			AnkrAdsNativeAndroid.GetTextureByteArray(testUnitId);
+		}
+
+		private void OnViewButtonClick()
+		{
+			const string testUnitId = "d396af2c-aa3a-44da-ba17-68dbb7a8daa1";
+			AnkrAdsNativeAndroid.ShowAd(testUnitId);
+		}
+		#endregion
+		
+		private void OnEnable()
 		{
-			base.DeActivateUseCase();
 			_ankrBannerAdImage.gameObject.SetActive(false);
 			_ankrBannerAdSprite.gameObject.SetActive(false);
+
+			_initializeButton.interactable = true;
+			_viewButton.interactable = false;
+			_loadButton.interactable = false;
+			_getTextureButton.interactable = false;
+
+			_initializeButton.onClick.AddListener(OnInitializeButtonClick);
+			_loadButton.onClick.AddListener(OnLoadButtonClick);
+			_viewButton.onClick.AddListener(OnViewButtonClick);
+			_getTextureButton.onClick.AddListener(OnGetTextureClick);
+		}
+
+		private void OnDisable()
+		{
+			_initializeButton.onClick.RemoveAllListeners();
+			_viewButton.onClick.RemoveAllListeners();
+			_loadButton.onClick.RemoveAllListeners();
+			UnsubscribeToCallbackListenerEvents();
+		}
+
+		private void ActivateNextButton(int buttonToActivate)
+		{
+			switch (buttonToActivate)
+			{
+				case 0:
+					_initializeButton.interactable = true;
```

Comment:
this can be simplified
_initializeButton.interactible = buttonToActivate == 0;
loadbutton.interactible = buttontoActivate == 1
etc